src/mongo_odm/model.hpp

```diff
@@ -24,8 +24,51 @@
 namespace mongo_odm {
 MONGO_ODM_INLINE_NAMESPACE_BEGIN
 
+/**
+ * Helper type trait widget that helps properly forward arguments to _id constructor.
+ */
+template <typename... Ts>
+struct Widget : public std::false_type {};
+
+template <typename T, typename T2, typename... Ts>
+struct Widget<T, T2, Ts...> : public std::is_same<T, std::decay_t<T2>> {};
+
+/**
+ * Base class for mongo_odm::model that provides _id and _id construction.
+ * TODO: Once CXX-940 is resolved, this model_odm_base will no longer be necessary and the argument
+ *       forwarding logic can take place in the mongo_odm::model class.
+ */
+template <typename IdType>
+class model_odm_base {
+   public:
+    template <typename... Ts, typename = std::enable_if_t<!Widget<model_odm_base, Ts...>::value>>
+    model_odm_base(Ts&&... ts) : _id(std::forward<Ts>(ts)...) {
+    }
+
+   protected:
+    IdType _id;
+};
+
+/**
+ * Specialization of mongo_odm::model base class for IdType=bsoncxx::oid that provides a default
+ * constructor for _id which safely constructs a bsoncxx::oid.
+ */
+template <>
+class model_odm_base<bsoncxx::oid> {
+   public:
+    template <typename... Ts, typename = std::enable_if_t<!Widget<model_odm_base, Ts...>::value>>
+    model_odm_base(Ts&&... ts) : _id(std::forward<Ts>(ts)...) {
```

Comment:
@hanumantmk PTAL at this ctor / inheritance templating :)
looks good to me though